subprojects/files/src/main/java/org/gradle/api/internal/file/archive/InputStreamAction.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.file.archive;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+ /**
+ * Functional interface to run an action against a {@link InputStream}
+ *
+ * @param <T> the action's result.
+ */
+public interface InputStreamAction<T> {
+    /**
+     * action to run against the passed {@link InputStream}.
+     */
+    T run(InputStream inputStream) throws IOException;
+}
```

Comment:
Wonder if we should reuse `IoFunction` here?

Reply:
Yeah, I think so - good idea.  It doesn't change the code much, and allows us to remove that class.